server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -284,72 +292,58 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
 
     protected abstract String contentType();
 
-    public static class MultiFields implements Iterable<Mapper> {
+    public static class MultiFields implements Iterable<FieldMapper> {
 
         public static MultiFields empty() {
-            return new MultiFields(ImmutableOpenMap.<String, FieldMapper>of());
+            return new MultiFields(Collections.emptyMap());
         }
 
         public static class Builder {
 
-            private final ImmutableOpenMap.Builder<String, Mapper.Builder> mapperBuilders = ImmutableOpenMap.builder();
+            private final Map<String, Function<BuilderContext, FieldMapper>> mapperBuilders = new HashMap<>();
```

Comment:
Yeah, I don't imagine ImmutableOpenMap is buying us too much. Might be worth `git blame` to see if there is anything back in the mists of time explaining it.

Reply:
I did some digging and this appears to have been there since the beginning, when ImmutableOpenMap was used all over the place.  I think it was initially supposed to be slightly more memory efficient, and the immutability was a nice add-on, but more modern JVMs and `Collections.immutableMap()` make that matter less.